### ISSUE TITLE
Custom jQuery not working on Multi-Part Forms

### ISSUE THREAD

**Issue by heartbreakkid58** [wpninjas/ninja-forms-multi-part#133]
I have used this event ready state, but this is ONLY works on first step.. I need to add different jQuery to all multi-form-parts.

`jQuery(document).on( 'nfFormReady', function( e, layoutView ) {
    // Your code goes here...
});`

[Custom jQuery not working on Wordpress Ninja Forms plugin
](https://stackoverflow.com/questions/40875677/custom-jquery-not-working-on-wordpress-ninja-forms-plugin)

**Comment by wpnzach** [comment]
jQuery works, but you must listen to the page change and re instantiate your script when the page changes.  The reCaptcha code in core is an example of this.  Closing as this is not a bug.

**Comment by arkid** [comment]
I understand that NF don't seem to want to provide any detailed support on this kind of query but is it possible to at least point anyone in such a situation in the right direction?  I've been testing with MutationObserver's in JS and I'm still getting no joy.

**Comment by husniazeez** [comment]
Yep same here as @arkid mentioned. Any luck @arkid ?

**Comment by arkid** [comment]
> Yep same here as @arkid mentioned. Any luck @arkid ?

I got this all working for the project I was doing.    I used a plugin, jquery-observe in the end which made the coding very simple to setup observable events.  

You can look at the source code of my page here:  https://beboilersmart.uk/quote/

Try to ignore the header, 100s of includes and general bloat on this page as its embedded within a rather messy WordPress theme!  My javascript is all embedded in the actual html of this page, it isn't an include so scroll down and you will see what I've done.

Basically I observe the next/back button wrapping div that Ninja Forms adds using 


```
jQuery('.quoteForm').observe('childlist subtree', '.nf-next-previous input[type="button"]', function(record) {
--
  | renderQuote();
  | });

```

And then I have this separate renderQuote function which gets called every time the user switches between the two parts of my multi-part form.     

The second part of this particular form has a textarea input field so I then check for the existence of that at the top of my function to determine which of the two parts we are in.

**Comment by husniazeez** [comment]
Thank you so much @arkid works like a charm 👍. Yes I agree extra elements we need to code beyond the ninja forms principals. Lets see any further ninja form releases come with extra tools.

**Comment by nilsringersma** [comment]
Is there really a need to observe? How about [this](https://gist.github.com/nilsringersma/c49da6383e58669978813b926a59e021) 😄  ?

**Comment by adriantee** [comment]
If there's anyone still struggling with this, you can try:
```
jQuery(document).on('nfFormReady', function() {
  console.log("ninja form ready");
  nfRadio.channel( 'nfMP' ).on('change:part', function() {
    console.log("[nfMP] change:part");
    // run your function here
  });
});
```
It works for Ninjaform 3.x

Reference: https://gist.github.com/nilsringersma/c49da6383e58669978813b926a59e021
